### PR TITLE
Record a reason for pod failure in EphemeralRunner

### PR DESCRIPTION
**Pull request by utam0k**
Users can see why pods failed.

Relate: https://github.com/actions/actions-runner-controller/issues/2888

### CODE REVIEW

File: controllers/actions.github.com/ephemeralrunner_controller.go [hunk 1]
```diff
@@ -423,12 +424,17 @@ func (r *EphemeralRunnerReconciler) cleanupRunnerLinkedSecrets(ctx context.Conte
 	return false, multierr.Combine(errs...)
 }
 
-func (r *EphemeralRunnerReconciler) markAsFailed(ctx context.Context, ephemeralRunner *v1alpha1.EphemeralRunner, log logr.Logger) error {
+func (r *EphemeralRunnerReconciler) markAsFailed(
```
Comment (nikola-jokic) [inline controllers/actions.github.com/ephemeralrunner_controller.go]: Can you please keep it in a single line? :relaxed:

Comment (utam0k) [reply controllers/actions.github.com/ephemeralrunner_controller.go]: Sure!

**Comment by utam0k** [comment]
@nikola-jokic Could you review this again? 🙏

**Review by nikola-jokic — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**